packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -55,33 +44,36 @@ const LinkControlSearchInput = ( {
 	}
 
 	return (
-		<form onSubmit={ selectSuggestionOrCurrentInputValue }>
-			<URLInput
-				className="block-editor-link-control__search-input"
-				value={ value }
-				onChange={ selectItemHandler }
-				onKeyDown={ ( event ) => {
-					if ( event.keyCode === ENTER ) {
-						return;
-					}
-					handleLinkControlOnKeyDown( event );
-				} }
-				onKeyPress={ handleLinkControlOnKeyPress }
-				placeholder={ __( 'Search or type url' ) }
-				__experimentalRenderSuggestions={ renderSuggestions }
-				__experimentalFetchLinkSuggestions={ fetchSuggestions }
-				__experimentalHandleURLSuggestions={ true }
-				__experimentalShowInitialSuggestions={ showInitialSuggestions }
-			/>
+		<form onSubmit={ handleSubmit }>
+			<div className="block-editor-link-control__search-input-wrapper">
+				<URLInput
+					className="block-editor-link-control__search-input"
+					value={ value }
+					onChange={ selectItemHandler }
+					placeholder={ __( 'Search or type url' ) }
+					__experimentalRenderSuggestions={ renderSuggestions }
+					__experimentalFetchLinkSuggestions={ fetchSuggestions }
+					__experimentalHandleURLSuggestions={ true }
+					__experimentalShowInitialSuggestions={ showInitialSuggestions }
+				/>
+
+				<Button
+					disabled={ ! value.length }
+					type="reset"
+					label={ __( 'Reset' ) }
+					icon="no-alt"
+					className="block-editor-link-control__search-reset"
+					onClick={ onReset }
+				/>
+			</div>
+			<div role="alert" aria-live="assertive">
```

Comment:
We should want to use the existing `speak` or `withSpokenMessages`:

- https://github.com/WordPress/gutenberg/tree/master/packages/a11y
- https://github.com/WordPress/gutenberg/tree/master/packages/components/src/higher-order/with-spoken-messages

That being said, this may all become unnecessary once #15745 is merged, which I am hoping to do sometime today.

Reply:
Shall we say we leave "as is" unless your PR gets merged? I'm conscious of doing work that will then be immediately undone.

Just to be clear, do we always prefer spoken messages over `aria-live="assertive"` and friends?

Reply:
> Shall we say we leave "as is" unless your PR gets merged? I'm conscious of doing work that will then be immediately undone.
> 
> Just to be clear, do we always prefer spoken messages over `aria-live="assertive"` and friends?

I think we should do one of:

- Remove it, in anticipation that Notice will handle this natively
- Or revise it to use `withSpokenMessages` or `speak`

I would not feel entirely confident about leaving it as it is, at least not without an accessibility review. It's not clear to me if there might be issues with one or both of (a) the fact that this is mounted after initial load or (b) if or how multiple `aria-live` in a page would interoperate.

Possibly relevant:

>Always make sure that the live region is present in the document first, and only then dynamically add/change any content.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions

Reply:
Ok I've added the `speak` and remove the custom implementation.

Reply:
Noting that the previously-mentioned #15745 is now merged. You shouldn't need to call `speak` at all, assuming the message as passed to `Notice` is the one to be spoken. You should remove the `useEffect` hook, else you risk double-announcing the message.

Also, I still see a separate `<div role="alert" aria-live="assertive">`. Not sure if this was removed at one point and possibly reintroduced in a rebase, but it shouldn't be there.